AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignIn/ConfirmDevice.swift

```
@@ -0,0 +1,110 @@
+                    deviceGroupKey: deviceMetadata.deviceGroupKey,
+                    deviceKey: deviceMetadata.deviceKey,
+                    password: deviceMetadata.deviceSecret)
+
+                let verifierData = Data(AmplifyBigIntHelper.getSignedData(
+                    num: devicePasswordVerifierConfig.passwordVerifier))
+                let saltData = Data(AmplifyBigIntHelper.getSignedData(
+                    num: devicePasswordVerifierConfig.salt))
+
+                let base64EncodedVerifier = verifierData.base64EncodedString()
+                let base64EncodedSalt = saltData.base64EncodedString()
+                let verifier = CognitoIdentityProviderClientTypes.DeviceSecretVerifierConfigType(
+                    passwordVerifier: base64EncodedVerifier,
+                    salt: base64EncodedSalt)
+                let input = ConfirmDeviceInput(
+                    accessToken: signedInData.cognitoUserPoolTokens.accessToken,
+                    deviceKey: deviceMetadata.deviceKey,
+                    deviceName: getCurrentDeviceName(),
+                    deviceSecretVerifierConfig: verifier)
+
+                let response = try await client.confirmDevice(input: input)
+                logVerbose("Successfully completed device confirmation with result \(response)",
+                           environment: environment)
+
+                // Save the device metadata to keychain
+                let credentialStoreClient = (environment as? AuthEnvironment)?.credentialStoreClientFactory()
+                let _ = try await credentialStoreClient?.storeData(
+                    data: .deviceMetadata(signedInData.deviceMetadata, signedInData.userName))
+                logVerbose("Successfully stored the device metadata in the keychain ",
+                           environment: environment)
+            }
+            catch {
+                logError("Failed to confirm the device \(error)",
```

Are we ignoring the error?

---

Yes.. This is the one I was talking about, that need to create a doc and passing back the response to the customer.